### PR TITLE
chore: update `super-linter` with new config

### PR DESCRIPTION
**Pull request by frytg**
Now provides a re-usable GitHub Actions workflow for all other repositories.

Note: The check will fail, since `main` does not contain the config files yet.

Example of a successful run using the latest dev version from this branch: https://github.com/swrlab/swr-prettier-config/runs/7746634766?check_suite_focus=true

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# docs: https://editorconfig.org
+# please make sure to keep this file in sync between repositories
+# to main file should be edited in `swr-prettier-config` and copied where needed
+
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_style = tab
```
Comment (rafaelmaeuer) [inline .editorconfig:10]: why ist default indent-style `tab` when all others are `space`?

File: .github/workflows/linter.yml [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        required: false
+        type: string
+        default: 'VERBOSE'
+      validate-all-codebase:
+        description: 'Whether to run validation on all files'
+        required: false
+        type: boolean
+        default: true
+
+env:
```
Comment (rafaelmaeuer) [inline .github/workflows/linter.yml:29]: very long list - everything necessary? - boilerplate?

File: .markdownlint.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+MD004: true
+
+MD007:
+  # Unordered list indentation
+  indent: 2
+
+# MD013/line-length - Line length
+MD013:
+  # Number of characters
+  line_length: 808
```
Comment (rafaelmaeuer) [inline .markdownlint.yaml:19]: 808 ?

File: docs/NPM.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  - [Publishing](#publishing)
+
+## Security
+
+Two-factor authentication (2FA) is enabled for all of SWR Audio Lab's packages (published as `@swrlab`).  
+To set up 2FA for your account, visit this guide: [docs.npmjs.com/configuring-two-factor-authentication](https://docs.npmjs.com/configuring-two-factor-authentication).
+
+## Publishing
+
+First, make sure you are logged into the `npm` CLI locally (`npm config list). If not, use the web authentication workflow:
```
Comment (rafaelmaeuer) [inline docs/NPM.md:16]:
```suggestion
First, make sure you are logged into the `npm` CLI locally (`npm config list`). If not, use the web authentication workflow:
```

File: .markdownlint.yaml (hunk 3 above)
Comment (frytg) [reply .markdownlint.yaml:19]: so far yes, there are some long code blocks that we still need to optimize

File: .github/workflows/linter.yml (hunk 2 above)
Comment (frytg) [reply .github/workflows/linter.yml:29]: yes, but this way we only have the list once

File: .editorconfig (hunk 1 above)
Comment (frytg) [reply .editorconfig:10]: JS and others are all tabs, only some space exclusions

**Pull request merged**